File: 100.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        
        def helper(n1, n2):
                        
            if not n1 and not n2:
                return True
            if n1 and not n2:
                return False
            if n2 and not n1:
                return False
            if n1.val != n2.val:
                return False
            return helper(n1.left, n2.left) and helper(n1.right, n2.right)

        
        return helper(p, q)
        

File: 100.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isSameTree(TreeNode* p, TreeNode* q) {
            std::stack<TreeNode*> st;
            st.push(p);
            st.push(q);

            while (!st.empty()) {
                TreeNode* cur_q = st.top();
                st.pop();
                TreeNode* cur_p = st.top();
                st.pop();

                if (!cur_p && !cur_q)
                    continue;
                if (!cur_p || !cur_q || cur_p->val != cur_q->val)
                    return false;

                st.push(cur_p->left);
                st.push(cur_q->left);
                st.push(cur_p->right);
                st.push(cur_q->right);
            }

            return true;
    }
};